### DISCUSSION TITLE
What does the statuses remove command do

### DISCUSSION THREAD

**Discussion by Tealk** [mastodon/mastodon#25488] [Q&A] [answered]
Hello all,

I would like to know what exactly the command `statuses remove` does, which posts are now deleted and which remain?

**Comment by trwnh** [comment]
Anything referenced locally or interacted with locally (faved, bookmarked, boosted, is a boost of a local user, mentions a local user, etc) will be kept. https://docs.joinmastodon.org/admin/tootctl/#statuses-remove

**Reply by Tealk** [reply]
No that's not the answer, because the statement there is way too vague.
What happens to bookmarks, boosts or stars?

It seems to be causing some problems: https://rollenspiel.social/@terraminima/110560857368239679

**Reply by trwnh** [reply, marked as answer]
https://github.com/mastodon/mastodon/blob/main/lib/mastodon/cli/statuses.rb

It finds all remote statuses, filters out the ones where the author is followed by a local user, or where they are replied to by a local status, or reblogs of local statuses, or statuses that have been reblogged locally, or pinned statuses, or where mentions exist, or where favourites exist, or where local bookmarks exist. then it deletes them

there may be an oversight for statuses which are *in reply to* a local status, as these are technically not interacted with, but i think the "mentions" check should exclude those...

the alternative explanation is that the missing replies may have been deleted by the remote users, for example, via the "automatically delete older statuses" feature (Settings > Automated post deletion)
